org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/ParameterizedTypeBinding.java

```diff
@@ -1062,6 +1062,24 @@ public boolean isHierarchyConnected() {
 		return this.superclass != null && this.superInterfaces != null;
 	}
 
+	@Override
+	public boolean hasUnconnectedSourceHierarchy() {
+		if (this.type.hasUnconnectedSourceHierarchy())
+			return true;
+		if (this.arguments != null) {
+			enterRecursiveFunction();
+			try {
+				for (TypeBinding argument : this.arguments) {
+					if (argument == null || argument.hasUnconnectedSourceHierarchy())
```

Comment:
For this to fully make sense, I may have to override `hasUnconnectedSourceHierarchy` also in other types like `TypeVariableBinding`, `WildcardBinding` etc.

Until then `enterRecursiveFunction` is more of a scarecrow than actually doing work, I suppose.

Reply:
this method no longer exists